test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system;
+
+import io.pravega.client.ClientFactory;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.ReaderConfig;
+import io.pravega.client.stream.ReinitializationRequiredException;
+import io.pravega.client.stream.impl.JavaSerializer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Abstract class that provides convenient event read/write methods for testing purposes.
+ */
+@Slf4j
+abstract class AbstractReadWriteTest {
```

Comment:
`AbstractFailoverTests` also performs reads and writes, should we have a common class for reading and writing? If you agree that this is possible and a good option, let's do the refactoring in a separate issue/PR.

Reply:
You are right, it could be interesting to encapsulate all the read/write functionality in a single abstract class. I will create a separate issue for refactoring system tests.